utf8proc.c

```diff
@@ -186,6 +186,37 @@ UTF8PROC_DLLEXPORT utf8proc_ssize_t utf8proc_encode_char(utf8proc_int32_t uc, ut
 }
 
 /* internal "unsafe" version that does not check whether uc is in range */
```

Comment:
The `tests.h` file is only used for parsing the input test files from Unicode consortium into UTF-8 strings.  The `GraphemeBreakTest.txt` file from the Unicode consortium contains (unpaired) surrogate "code points", presumably for testing UTF-16 code.   The test program encodes these into (invalid) UTF-8, and then `utf8proc_map` (correctly) signals a `UTF8PROC_ERROR_INVALIDUTF8` error.

The graheme test doesn't use 0xFFFF or 0xFFFE, as far as I can tell.  I'm not sure why these cases were included in the `encode` function.  (Maybe some kind of BOM thing?  Not that BOM is needed or recommended for UTF-8 data.)